soc/arm/nordic_nrf/common/poweroff.c

```diff
@@ -6,11 +6,19 @@
 #include <zephyr/sys/poweroff.h>
 #include <zephyr/toolchain.h>
 
+#if defined(SOC_SERIES_NRF51X) || defined(SOC_SERIES_NRF52X)
 #include <hal/nrf_power.h>
+#else
+#include <hal/nrf_regulators.h>
+#endif
 
 void z_sys_poweroff(void)
 {
+#if defined(SOC_SERIES_NRF51X) || defined(SOC_SERIES_NRF52X)
```

Comment:
@anangl ?

Reply:
IMHO, `HAS_HW_*` should really be `DT_HAS....`, Zephyr needs to take its hw description from DT.

Reply:
Ok so you could use `#if DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_regulators)` etc

Reply:
> wrt portability to potential new series, it could perfectly be that they require new APIs or more calls, so I prefer the ifdeffery based on SOC series for now.

Yeah what I'm saying is that to make sure you don't miss that with the new soc port you may go for something like

```c
#if defined(SOC_SERIES_NRF51X) || defined(SOC_SERIES_NRF52X)
...
#elif defined(SOC_SERIES_NRF53)
...
#else
#error "I have no idea how to power off"
#endif
```

It's not a big deal, just thinking it may be less likely to be forgotten, a bit more defensive maybe.

Reply:
hmm the `#error` branch is unlikely, though, because the series option needs to explicitely select `HAS_POWEROFF`

Reply:
Cool, lost track of the `SOC` vs `DT_HAS` so I guess it's between you and @anangl